### PR TITLE
✨ 🏷️  369 - add create participant flow to consent-api, update req/res types

### PR DESCRIPTION
**Pull request by anncatton**
## Description of Changes

Adds create Participant flow to Consent API, and corrects the data being returned from the Data Mapper for this flow.

### Consent API
- adds `POST /participants` endpoint with request validation and error handling
- adds `CreateParticipant` request and response schemas to swagger
- adds `createParticipant` service call
- adds test file

### Data Mapper
- fixes data object returned from `createParticipant` service function and consent-das `createParticipantConsentData` function

### Types
- Adds request and response types for `CreateParticipant` in `consent-api` service dir
- refactors `ConsentParticipantBase` slightly: removes `ConsentToBeContacted` as this type was the `literal: true` value that should only be used in request types, not the base type. This field is now added separately as needed
- modifies `CreateParticipant` response type for data-mapper to reflect change in base types
- removes `previousLifecycleState` from `ConsentParticipant` as it's not part of the create flow; can be added to the update participant types when needed. Thinking here is that the create participant flow collects the minimum amount of participant info, so the response type can be a subset of the whole participant object (incl both PI and Consent DAS data)

## PR Readiness Checklist

- [x] "Expected Outcome(s)" in ticket have been met
- [x] Ticket number included in PR title
- [x] Connected ticket to PR
- [x] Labels added to PR for service name (`consent-api`, `data-mapper`, etc...), type (`chore`, `documentation`, etc...), status (`draft`, `on-hold`, etc...) **if applicable**
- [x] Manual testing completed
- [x] Builds locally without errors or warnings
- [x] Tests are updated (if required) and passing
- [ ] PR feedback has been addressed **if applicable**
- [ ] New environment variables added to `.env.schema` files, `README.md`
- [x] Added copyrights to new files

### CODE REVIEW

File: apps/consent-api/tests/routers/participants.spec.ts [hunk 1]
```diff
@@ -0,0 +1,94 @@
...
+		const appConfig = getAppConfig();
+		const response = await request(App(appConfig))
+			.post('/participants')
+			.send({ data: mocks.createParticipantRequest });
+
+		expect(response.status).toEqual(201);
+		expect(response.body).toStrictEqual(mocks.createParticipantResponse);
+	});
+
+	it('Invalid request - false consentToBeContacted should return RequestValidationError', async () => {
```
Comment (mistryrn) [inline apps/consent-api/tests/routers/participants.spec.ts:81]: Very helpful description 👏

File: apps/data-mapper/src/services/create.ts [hunk 2]
```diff
@@ -111,8 +111,8 @@ export const createParticipant = async ({
 		}
 		// parse complete result from both dases
 		const result = CreateParticipantResponse.safeParse({
-			...participantPiData,
-			...participantConsentData,
+			...participantPiData.data,
```
Comment (mistryrn) [inline apps/data-mapper/src/services/create.ts:114]: Nice catch.

File: packages/types/src/entities/Participant.ts [hunk 3]
```diff
@@ -82,20 +82,18 @@ export const PIParticipantBase = ParticipantIdentityBase.merge(
...
+	consentGroup: ConsentGroup,
+	emailVerified: z.boolean(),
+	isGuardian: z.boolean(),
+});
 export type ConsentParticipantBase = z.infer<typeof ConsentParticipantBase>;
 
 export const ConsentParticipant = ConsentParticipantBase.merge(
 	z.object({
 		id: NanoId,
-		previousLifecycleState: LifecycleState.optional(),
```
Comment (mistryrn) [inline packages/types/src/entities/Participant.ts:98]: Is this `previousLifecycleState` no longer needed here?

**Review by mistryrn — COMMENTED** [review]
Lots of good stuff in here! Just added some comments/questions for my own clarification as it's been a while since I've touched the Consent API layer 🙂

File: apps/consent-api/src/services/create.ts [hunk 4]
```diff
@@ -85,3 +90,48 @@ export const createInvite = async (
...
+		const { data } = await axiosClient.post(urlJoin(dataMapperUrl, 'participants'), request);
+		const participant = CreateParticipantResponse.safeParse(data);
+
+		if (!participant.success) {
+			logger.error(
+				'POST /participants',
+				'Received invalid data in response.',
+				participant.error.issues,
+			);
+			return failure('SYSTEM_ERROR', participant.error.message);
```
[the remark this answers is not in the document]
Comment (mistryrn) [reply apps/consent-api/src/services/create.ts:115]: Ah, that makes sense. Thanks for reminding me!

File: packages/types/src/entities/Participant.ts (hunk 3 above)
Comment (mistryrn) [reply packages/types/src/entities/Participant.ts:98]: Got it, thanks! I'm okay to proceed with this PR without updating the naming but that is a good point. Any suggestions for a better name? We can always update it in a follow-up PR

**Review by mistryrn — DISMISSED** [review]
Thanks for answering my questions! This looks good to me 🙂

File: packages/types/src/entities/Participant.ts (hunk 3 above)
Comment (anncatton) [reply packages/types/src/entities/Participant.ts:98]: not off the top of my head but i'll put some thought to it 🤔

**Review by mistryrn — APPROVED** [review]

**Pull request merged**